cvmfs/catalog.h

```diff
@@ -163,7 +163,7 @@ class Catalog : public SingleCopy {
   void TakeDatabaseFileOwnership();
   void DropDatabaseFileOwnership();
   bool OwnsDatabaseFile() const {
-    return NULL != database_ && database_->OwnsFile();
+    return ((database_ != NULL) && database_->OwnsFile()) || managed_database_;
```

Comment:
Can we just use `database_->OwnsFile()` and avoid double-bookkeeping of the state in `managed_database_`?